src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -387,6 +387,8 @@
         Ref("LessThanOrEqualToSegment"),
         Ref("NotEqualToSegment"),
         Ref("LikeOperatorSegment"),
+        Sequence("IS", "DISTINCT", "FROM"),
+        Sequence("IS", "NOT", "DISTINCT", "FROM"),
```

Comment:
Would it make sense to turn this into a Grammar or Segment of its own?

Reply:
Possibly. We tend to do that for a few reasons:
1) To avoid repeating lots of same code in lots of places (not the case here - though admittedly same code also in Postgres dialect)
2) To allow it to overridden in other dialects (not the case here)

As neither is really the case here, I'd leave it as is. Having another Grammar/Segment adds a level of redirection and don't think that's necessary here so clearer to have the actual grammar.

BTW if interested in contributing, then this is a good read: https://github.com/sqlfluff/sqlfluff/wiki/Contributing-Dialect-Changes

Reply:
BTW we could also consolidate this down to one statement:

```py
Sequence("IS", Ref.keyword("NOT", optional=True), "DISTINCT", "FROM"),
```

But again, think it's slightly clearer as two statements.

Always lots of ways of doing things in SQLFluff!